src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -2254,6 +2254,14 @@ internal void OnFedAuthInfo(SqlFedAuthInfo fedAuthInfo)
                 {
                     // GetFedAuthToken should have updated _newDbConnectionPoolAuthenticationContext.
                     Debug.Assert(_newDbConnectionPoolAuthenticationContext != null, "_newDbConnectionPoolAuthenticationContext should not be null.");
+
+                    if (_newDbConnectionPoolAuthenticationContext != null)
+                    {
+                        // Try adding this new _newDbConnectionPoolAuthenticationContext to the _dbConnectionPool's AuthenticationContextKeys if it is not in there yet.
```

Comment:
Please stop adding unncessary comments. This causes code bloat, and comments in code can easily fall out of date, leaving confusion to the next developer.
This is not best practice. Please put comments in your commits if you want to convey something and commit often. 
Comment only if the logic that is happening is not obvious or difficult to understand.

Reply:
Removed.